src/sentry/api/endpoints/artifact_lookup.py

```diff
@@ -0,0 +1,154 @@
+import logging
+
+from django.http import Http404, HttpResponse, StreamingHttpResponse
+from rest_framework.request import Request
+from rest_framework.response import Response
+from symbolic import SymbolicError, normalize_debug_id
+
+from sentry import ratelimits
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint, ProjectReleasePermission
+from sentry.api.endpoints.debug_files import has_download_permission
+from sentry.api.serializers import serialize
+from sentry.models import DebugIdArtifactBundle, File
+
+logger = logging.getLogger("sentry.api")
+
+
+@region_silo_endpoint
+class ArtifactLookupEndpoint(ProjectEndpoint):
+    permission_classes = (ProjectReleasePermission,)
+
+    def download_file(self, file_id, project):
+        rate_limited = ratelimits.is_limited(
+            project=project,
+            key=f"rl:DSymFilesEndpoint:download:{file_id}:{project.id}",
+            limit=10,
+        )
+        if rate_limited:
+            logger.info(
+                "notification.rate_limited",
+                extra={"project_id": project.id, "file_id": file_id},
+            )
+            return HttpResponse({"Too many download requests"}, status=403)
```

Comment:
Maybe `429` instead?

Reply:
I copied all this from the `DebugFilesEndpoint`. No idea why that is using `403`? Either way, I changed this to `429` as you suggested.